### PR TITLE
Switch to Lucene's new IntField/LongField/FloatField/DoubleField.

### PR DESCRIPTION
**Pull request by jpountz**
Lucene introduced new numeric fields that index both points and doc values. This has the same semantics as indexing one field for points and another one for doc values as we did before, but covering both data structures in a single field yielded a speedup in Lucene's nightly benchmarks (see annotation [AH](http://people.apache.org/~mikemccand/lucenebench/sparseResults.html#index_throughput)) which would be interesting to get too.

This commit does not switch to factory methods for queries such as `LongField#newRangeQuery` for now, we'll need to look into it in a follow-up.

### CODE REVIEW

**Review by benwtrent — COMMENTED** [review]
We should add to `UnsignedLongFieldMapper` as well I think. It also has doc values and is indexed.

File: server/src/main/java/org/elasticsearch/index/mapper/DateFieldMapper.java [hunk 1]
```diff
@@ -907,12 +908,15 @@ protected void parseCreateField(DocumentParserContext context) throws IOExceptio
...
+        } else if (hasDocValues) {
             context.doc().add(new SortedNumericDocValuesField(fieldType().name(), timestamp));
-        } else if (store || indexed) {
+        } else if (indexed) {
+            context.doc().add(new LongPoint(fieldType().name(), timestamp));
+            // When the field doesn't have doc values so that we can run exists queries, we also need to index the field name separately.
+            context.addToFieldNames(fieldType().name());
+        } else {
             context.addToFieldNames(fieldType().name());
         }
```
[the remark this answers is not in the document]
Comment (jpountz) [reply server/src/main/java/org/elasticsearch/index/mapper/DateFieldMapper.java]: Thanks for catching this, I had in my mind that we should add to field names whenever doc values is false, but this field and others do indeed only do this if the field is either indexed or stored as well.

**Review by benwtrent — APPROVED** [review]

**Review by javanna — APPROVED** [review]
LGTM

**Comment by javanna** [comment]
the bwc test failure is one that has happened before, does not look like it's caused by your test, see #92058 . The percolator failure seems legit :)

**Comment by jpountz** [comment]
> The percolator failure seems legit

Absolutely, it's caused by https://github.com/apache/lucene/pull/12109. The test failure should go away when we upgrade to the final release.

**Comment by javanna** [comment]
@elasticmachine  update-branch

**Comment by javanna** [comment]
@elasticmachine update branch

**Comment by javanna** [comment]
@jpountz you were right about the percolator test failure, also auto-merge for the win here :)

**Pull request merged**